padertorch/ops/losses/regression.py

```diff
@@ -277,5 +277,21 @@ def log1p_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
         reduction=reduction
     )
 
-# TODO: Add log1p_mse loss from interspeech paper
+
+def source_aggregated_sdr_loss(
+        estimate: torch.Tensor,
+        target: torch.Tensor,
+) -> torch.Tensor:
+    """
+    The source-aggregated SDR loss. There is no `reduction` argument because the loss always
+    combines the losses for multiple paris of targets and estimates.
+    """
+    # Calculate the source-aggregated SDR: Sum the squares of all targets and
+    # all errors before computing the ratio.
+    sa_sdr = 10 * torch.log10(
+        torch.sum(target**2) / torch.sum((estimate - target)**2)
```

Comment:
Question: Is mean more stable?

Reply:
I don't think that it makes a difference. I just tested with very large input shapes and input values and couldn't observe a difference larger than 1e-6.

Reply:
ok, great. I am not sure, if an overflow may be a problem for float32, but I also don't know if the mean is overflow stable.